src/stream/min_by.rs

```diff
@@ -0,0 +1,54 @@
+use std::cmp::Ordering;
+use std::pin::Pin;
+
+use super::stream::Stream;
+use crate::future::Future;
+use crate::task::{Context, Poll};
+
+/// A future that yields the minimum item in a stream by a given comparison function.
+#[derive(Clone, Debug)]
+pub struct MinBy<S: Stream, F> {
+    stream: S,
+    compare: F,
+    min: Option<S::Item>,
+}
+
+impl<S: Stream + Unpin, F> Unpin for MinBy<S, F> {}
+
+impl<S: Stream + Unpin, F> MinBy<S, F> {
```

Comment:
Instead of requiring `Unpin`, we should use `unsafe_pinner` like for the `Take` struct.